ocaml/testsuite/tests/typing-modes/syntax-error.compilers.reference

```diff
@@ -0,0 +1,123 @@
+Line 5, characters 27-28:
+5 | let local_ foo : string @@ = "hello";;
+                               ^
+Error: Syntax error
+Line 2, characters 17-18:
+2 | let local_ foo @ = "hello";;
+                     ^
+Error: Syntax error
+Line 2, characters 34-35:
+2 | let local_ foo : 'a. 'a -> 'a @@  = fun x -> x;;
+                                      ^
+Error: Syntax error
+Line 2, characters 29-30:
+2 | let foo : type a. a -> a @@  = fun x -> x;;
+                                 ^
+Error: Syntax error
+Line 2, characters 13-14:
+2 | let (x, y) @ = "hello", "world";;
+                 ^
+Error: Syntax error
+Line 2, characters 18-19:
+2 | let (x, y) : _ @@ = "hello", "world";;
+                      ^
+Error: Syntax error
+Line 2, characters 10-11:
+2 | let foo @ = "hello";;
+              ^
+Error: Syntax error
+Line 2, characters 26-27:
+2 | let foo = ("hello" : _ @@ );;
+                              ^
+Error: Syntax error
+Line 2, characters 21-22:
+2 | let foo = ("hello" @ );;
+                         ^
+Error: Syntax error
+val foo : bar:string -> string = <fun>
+Line 4, characters 19-20:
+4 |   foo ~(bar : _ @@ )
+                       ^
+Error: Syntax error
+Line 4, characters 14-15:
+4 |   foo ~(bar @ )
```

Comment:
Many of these errors (e.g. all of these ones) do not in fact show improved error messages. Do you know why that is?

Reply:
OK I think it's because of inlining. I removed some inlining and it's giving better messages.